File: scripts/check-versions.ts
#!/usr/bin/env bun
/** Проверяет, что версия одинакова во всех местах, где она записана. */

import { readVersions } from "./versions.ts";

const found = readVersions();
const distinct = new Set(found.map((f) => f.version));
if (distinct.size !== 1 || found.some((f) => !f.version)) {
  console.error("Версии расходятся:");
  for (const f of found) console.error(`  ${f.file}: ${f.version ?? "не найдена"}`);
  console.error("Исправьте вручную или выпустите версию через `bun run release`.");
  process.exit(1);
}

File: scripts/check_versions.py
"""Проверяет, что версия одинакова во всех местах, где она записана, включая uv.lock."""

from __future__ import annotations

import sys

from versions import LOCK, FoundVersion, lock_version, read_versions, utf8_output


def main() -> int:
    utf8_output()
    found = [*read_versions(), FoundVersion(LOCK, lock_version())]
    if len({f.version for f in found}) != 1 or any(not f.version for f in found):
        print("Версии расходятся:", file=sys.stderr)
        for f in found:
            print(f"  {f.file}: {f.version or 'не найдена'}", file=sys.stderr)
        print(
            "Исправьте вручную (uv.lock обновляет `uv lock`) или выпустите версию через "
            "`uv run python scripts/release.py`.",
            file=sys.stderr,
        )
        return 1
    return 0


if __name__ == "__main__":
    sys.exit(main())
